### PR TITLE
MES-2836: Advanced Search

### PR DESCRIPTION
**Pull request by szabolcs-keresztes**
Implementing Advance search UI
Linking it with Provider
Linking results to Results list
Loading Spinner on button

IMG_851EBBE0B020-1

### CODE REVIEW

**Comment by ammarhaiderbjss** [comment]
@szabolcs-keresztes quick screenshot?

File: config.xml [hunk 1]
```diff
@@ -96,5 +96,5 @@
     <plugin name="cordova-plugin-network-information" spec="^2.0.1" />
     <plugin name="cordova-plugin-device-authentication" spec="https://github.com/dvsa/cordova-plugin-device-authentication" />
     <plugin name="cordova-plugin-appconfig" spec="1.1.4" />
-    <engine name="ios" spec="4.5.5" />
+    <engine name="ios" spec="^4.5.5" />
```
Comment (ammarhaiderbjss) [inline config.xml]: can we keep this fixed?

File: src/pages/test-results-search/test-results-search.ts [hunk 2]
```diff
@@ -78,4 +81,22 @@ export class TestResultsSearchPage extends BasePageComponent {
     }
   }
 
+  advancedSearch(advancedSearchParams: AdvancedSearchParams): void {
+    this.showAdvancedSearchSpinner = true;
+    this.searchProvider.advancedSearch(advancedSearchParams)
+      .pipe(
+        tap(data => console.log('Advanced Search', JSON.stringify(data))),
```
Comment (ammarhaiderbjss) [inline src/pages/test-results-search/test-results-search.ts]: remove this log?

File: src/pages/test-results-search/test-results-search.ts [hunk 3]
```diff
@@ -78,4 +81,22 @@ export class TestResultsSearchPage extends BasePageComponent {
...
+        map((result) => {
+          this.searchResults = result;
+          this.showAdvancedSearchSpinner = false;
+        }),
+        catchError((err) => {
+          this.showAdvancedSearchSpinner = false;
+          return of(console.log('ERROR', JSON.stringify(err)));
+        }),
+      )
+      .subscribe();
```
Comment (ammarhaiderbjss) [inline src/pages/test-results-search/test-results-search.ts]: Can we use the `| async` pipe in the template to render this? That will also handle unsubscribing

File: config.xml (hunk 1 above)
Comment (szabolcs-keresztes) [reply config.xml]: yes

File: src/pages/test-results-search/test-results-search.ts (hunk 2 above)
Comment (szabolcs-keresztes) [reply src/pages/test-results-search/test-results-search.ts]: removed

File: src/pages/test-results-search/test-results-search.ts (hunk 3 above)
Comment (szabolcs-keresztes) [reply src/pages/test-results-search/test-results-search.ts]: Would be good. But let's keep that as an improvement on top of this.

File: src/pages/test-results-search/components/advanced-search/advanced-search.html [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        <ion-datetime
+          displayFormat="DD/MM/YYYY"
+          placeholder="select a date"
+          (ionChange)="endDateChanged($event)"></ion-datetime>
+      </ion-item>
+    </ion-col>
+  </ion-row>
+  <ion-row>
+    <button [attr.disabled]="showSpinner ? 'true' : null" class="mes-primary-button search-tests" ion-button (click)="searchTests()">
+      <span class="search-tests-text">Search tests</span>
```
Comment (buzzub) [inline src/pages/test-results-search/components/advanced-search/advanced-search.html]: The buttons in the app normally wrap a `<h3>` element to style them as bold.

File: src/pages/test-results-search/test-results-search.ts [hunk 5]
```diff
@@ -62,14 +65,16 @@ export class TestResultsSearchPage extends BasePageComponent {
...
+      this.showSearchSpinner = true;
       this.searchProvider.applicationReferenceSearch(this.candidateInfo)
       .pipe(
         tap(() => this.hasSearched = true),
         map((results) => {
           this.searchResults = results;
           this.showSearchSpinner = false;
         }),
-        catchError(() => {
+        catchError((error) => {
```
Comment (buzzub) [inline src/pages/test-results-search/test-results-search.ts]: `error` param isn't used

File: src/pages/journal/components/test-results-search-card/test-results-search-card.ts [hunk 6]
```diff
@@ -12,6 +12,7 @@ export class TestResultsSearchCardComponent {
   constructor(private navController: NavController) { }
 
   navigateToTestResultsSearch = () => {
+    console.log('navigating to ', TEST_RESULTS_SEARCH_PAGE);
```
Comment (buzzub) [inline src/pages/journal/components/test-results-search-card/test-results-search-card.ts]: Left over logging?

File: src/pages/test-results-search/components/advanced-search/advanced-search.html (hunk 4 above)
Comment (szabolcs-keresztes) [reply src/pages/test-results-search/components/advanced-search/advanced-search.html]: Changed it in both tabs

File: src/pages/test-results-search/test-results-search.ts (hunk 5 above)
Comment (szabolcs-keresztes) [reply src/pages/test-results-search/test-results-search.ts]: removed

File: src/pages/journal/components/test-results-search-card/test-results-search-card.ts (hunk 6 above)
Comment (szabolcs-keresztes) [reply src/pages/journal/components/test-results-search-card/test-results-search-card.ts]: removed

**Pull request merged**